### PR TITLE
Issue1028-benchmark-tutorial

### LINKED ISSUE #1028: Resurrect test_2d_benchmark_geiger.py

### PR DESCRIPTION
**Pull request by IvarStefansson**
## Proposed changes

Resurrects the regular benchmark setup as a tutorial. 

I discovered that the plot_2d option of plot_grid now works for mdgs as long as no glyphs are used. I rolled out in tutorials, obtaining way more beautiful plots. Finally a small improvement in naming of two ad array wrapper functions.

Resolves #1028.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality. At least sort of not.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: tutorials/benchmark_simulation.ipynb [hunk 1]
```diff
@@ -0,0 +1,305 @@
+{
```
Comment (keileg) [inline tutorials/benchmark_simulation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.            # See section on scaling for explanation of the conversion.
</pre><p>This was not clear to me.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1071/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: tutorials/grid_topology.ipynb [hunk 2]
```diff
@@ -50,19 +50,11 @@
    "execution_count": 2,
```
Comment (keileg) [inline tutorials/grid_topology.ipynb:50]: <p>Can we avoid this error?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1071/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

**Review by keileg — APPROVED** [review]
Two minor questions; please handle as you see fit before merging.

File: tutorials/benchmark_simulation.ipynb (hunk 1 above)
Comment (IvarStefansson) [reply tutorials/benchmark_simulation.ipynb:1]: Sorry, forgot to delete after copying.

File: tutorials/grid_topology.ipynb (hunk 2 above)
Comment (IvarStefansson) [reply tutorials/grid_topology.ipynb:50]: Sorry, notebook outputs and git aren't the best of friends.

**Pull request merged**